assets/javascripts/discourse/components/ai-persona-editor.gjs

```diff
@@ -34,10 +35,43 @@ export default class PersonaEditor extends Component {
   @tracked editingModel = null;
   @tracked showDelete = false;
 
+  // this is a bit awkward but the model is not tracked correctly
+  @tracked visionEnabled = false;
+  @tracked maxPixelsValue = null;
+
   @action
   updateModel() {
     this.editingModel = this.args.model.workingCopy();
+    this.visionEnabled = this.editingModel.vision_enabled;
     this.showDelete = !this.args.model.isNew && !this.args.model.system;
+    this.maxPixelsValue = this.findClosestPixelValue(
+      this.editingModel.vision_max_pixels
+    );
+  }
+
+  findClosestPixelValue(pixels) {
+    let value = "high";
+    this.maxPixelValues.forEach((info) => {
+      if (pixels === info.pixels) {
+        value = info.id;
+      }
+    });
+    return value;
+  }
+
+  get maxPixelValues() {
```

Comment:
Since the value of this never changes, you can add `@cached` to the getter with `
import { cached } from '@glimmer/tracking';` . Then you can remove the `this._maxPixelValues` check

Reply:
done